proposals/0000-reconsider-typealias-in-protocol.md

```diff
@@ -0,0 +1,138 @@
+introduces counterproductive situations that often have no practical value and are hard to judge and assess. 
+
+```swift
+protocol P {
+    associatedtype E
+    func foo(_ arg: E)
+}
+protocol P1: P {...} 
+
+class Foo: P {
+    typealias E = String
+
+    func foo(_ arg: E) {}
+}
+// This will break Foo and any other conformances to P and its descendants. 
+extension P { typealias E = Bool } 
+```
+A further contradiction is seen with the convention of leaving out any implementation details within a protocol body:
+those are ought to be described in an extension or upon conformance.
+
+### Type aliases in extensions
+Currently a `typealias` in a protocol extension that conflicts with an `associatedtype` is treated as a same-type constraint on the associatedtype **if** there are no appliances of the type name **among the requirements**. Example:
+
+```swift
+protocol P {associatedtype A}
+extension P {typealias A = Int} // same-type constraint on A.
+```
+
+If there **is** an appliance of `A`, an error occurs:
+
+```swift
+protocol P {
+  associatedtype A
+  func foo() -> A // 'A' is ambiguous for type lookup in this context
+}
+extension P {typealias A = Int}
+```
+
+This is inconsistent with the behavior of any other implementation details declared in protocol extensions, which become defaults. The fact that an appliance generates an error is either a bug or proof that the rules aren't solid and conflict with the language on the implementation level.
+
+## Proposed solution
+
+**Swift 4.1** began preparations to eliminate the feature by adding certain warnings and conditionally disallowing such type aliases in protocol extensions by raising a `type look up ambiguity` error towards appliances of the conflicting type name, if any.
+
+#### `#1`
+```swift
+protocol P {associatedtype A}
+protocol P1: P {typealias A = Int}
+// warning: typealias overriding associated type 'R' from protocol 'P' is better expressed as same-type constraint on the protocol
+```
+#### `#2`
+```swift
+protocol P {typealias A = Int}
+protocol P1: P {associatedtype A}
+// warning: associated type 'R' is redundant with type 'R' declared in inherited protocol 'P'
+```
+#### `#3`
+```swift
+protocol P {
+  associatedtype A
+  func foo() -> A // error: 'A' is ambiguous for type lookup in this context
+}
+extension P {typealias A = Int}
+```
+### Solution
+Now that we have powerful and expressive where clauses for protocol declarations,
+I propose to bring consistency to the usage of type aliases within protocols and protocol extensions.
+
+* [Type aliases in protocol extensions](#type-aliases-in-protocol-extensions). There are two options:
+    * Permit type aliases to act as default values for associated types in protocol extensions. This is the preferred           solution. 
+    * Change the error message to propose the current syntax instead of simply producing **#3**,
+    which is actually a generic error.
```

Comment:
A proposal should provide a specific change, not a choice between two options. Since you prefer the first option (permit type aliases to act as default values), I recommend stating that as the proposed semantic change. The second option can go into an "alternatives considered" section at the end.